src/generateGraphData.ts

```diff
@@ -0,0 +1,133 @@
+export interface ICommit {
+  sha: string;
+  parents: string[];
+}
+export interface INode {
+  sha: string;
+  dot: { lateralOffset: number; branch: number };
+  routes: IRoute[];
+  yOffset: number;
+}
+export interface IRoute {
+  from: number;
+  to: number;
+  branch: number;
+}
+
+const Node = (
+  sha: string,
+  offset: number,
+  branch: number,
+  routes: IRoute[],
+  yOffset: number
+): INode => ({
+  sha,
+  dot: { lateralOffset: offset, branch },
+  routes,
+  yOffset
+});
+
+const remove = function (list: number[], item: number) {
+  list.splice(list.indexOf(item), 1);
+  return list;
+};
```

Comment:
Let's name the function directly.

```suggestion
function remove(list: number[], item: number): number[] {
  list.splice(list.indexOf(item), 1);
  return list;
};
```